spec-0012/index.md

```diff
@@ -0,0 +1,185 @@
+---
+title: "SPEC 12 — Formatting mathematical expressions"
+number: 12
+date: 2024-06-06
+author:
+  - "Pamphile Roy <roy.pamphile@gmail.com>"
+  - "Matt Haberland <mhaberla@calpoly.edu>"
+discussion: https://discuss.scientific-python.org/t/spec-12-formatting-mathematical-expressions
+endorsed-by:
+---
+
+## Description
+
+[PEP 8](https://peps.python.org/pep-0008)
+and other established styling documents do not include guidelines about
+styling mathematical expressions. This leads to individual interpretation and
+styles which may conflict with those of others. We seek to standardizing the
+way we represent mathematics for the same reason we standardize other code:
+it brings consistency to the ecosystem and allows collaborators to focus on
+more important aspects of the code.
+
+This SPEC standardize the formatting of mathematical expressions.
+
+## Terminology
+
+An "explicit" expression is a code expression enclosed within parentheses or
+otherwise syntactically separated from other expressions (i.e. by code other
+than operators, whitespace, literals, or variables). For example, in the list
+comprehension:
+```python3
+for j in range(1, i + 1)
+```
+The output expression `j` is one explicit expression and the input sequence
+`range(1, i + 1)` is another. 
+
+A "subexpression" is subset of an expression that is either explicit or could
+be made explicit (i.e. with parentheses) without affecting the order of
+operations. In the example above, `j` and `range(1, i + 1)` can also be
+referred to as explicit subexpressions of  the whole expression, and `1` and
```

Comment:
"...of  the whole expression"

Assuming I'm reading this correctly, I would be careful here as `for j in range(1, i + 1)` is not an "expression" in the sense of the Python AST. `for` is a kind of "statement".